ERCS/erc-7837.md

```diff
@@ -0,0 +1,412 @@
+---
+eip: 7837
+title: Diffusive Tokens
+description: A fungible token that mints new tokens on transfer, charges a per-token native fee, and enforces a capped supply.
+author: James Savechives (@jamesavechives)
+discussions-to: https://ethereum-magicians.org/t/erc-7837-diffusive-tokens/21989
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-12-07
+---
+
+## Abstract
+
+This ERC proposes a standard for a new type of fungible token, called **Diffusive Tokens (DIFF)**. Unlike traditional [ERC-20](./eip-20.md) tokens, transferring DIFF tokens does not decrease the sender’s balance. Instead, it *mints* new tokens directly to the recipient, increasing the total supply on every transfer action. A fixed native currency fee is charged per token transferred, and this fee is paid by the sender to the contract owner. The supply growth is limited by a maximum supply set by the owner. Token holders can also burn their tokens to reduce the total supply. These features enable a controlled, incentivized token distribution model that merges fungibility with a built-in economic mechanism.
+
+## Motivation
+
+Traditional [ERC-20](./eip-20.md) tokens maintain a constant total supply and simply redistribute balances on transfers. While this model is widespread, certain use cases benefit from a token design that continuously expands supply during transfers, simulating a controlled "diffusion" of value. The Diffusive Token model may be suitable for representing claims on real-world goods (e.g., a product batch like iPhone 15 units), digital goods, or controlled asset distributions where initial token distribution and ongoing availability need to be managed differently.
+
+This model also includes a native currency fee per token transferred, incentivizing careful, value-driven transfers and providing a revenue stream for the token’s issuer. The maximum supply cap prevents unbounded inflation, ensuring long-term scarcity. The ability for owners to burn tokens to redeem underlying goods or services directly maps on-chain assets to real-world redemptions.
+
+## Specification
+
+### Terminology
+
+- **Diffusive Token**: A fungible token unit that is minted on transfers.
+- **Max Supply**: The maximum total supply the token can reach.
+- **Transfer Fee**: A fee in native blockchain currency (e.g., ETH) that must be paid by the sender for each token transferred. The total fee = `transferFee * amount`.
+- **Burn**: The action of destroying tokens, reducing both the holder’s balance and the total supply.
+
+### Data Structures
+
+- **Balances Mapping**: `balances[address] = uint256` tracks the number of tokens owned by each address.
+  
+  ```solidity
+  mapping(address => uint256) private balances;
+  ```
```

Comment:
This is an implementation detail. Only specify the externally visible behaviour of the contract (like a Solidity interface.)